pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -189,38 +189,40 @@ func withNetworkIfacesResources(
 		return f(vmi, domainSpec)
 	}
 
-	domainSpecWithIfacesResource := appendPlaceholderInterfacesToTheDomain(vmi, domainSpec)
-	dom, err := f(vmi, domainSpecWithIfacesResource)
+	originalIfaces := appendPlaceholderInterfacesToTheDomain(vmi, domainSpec)
+	dom, err := f(vmi, domainSpec)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(domainSpec.Devices.Interfaces) == len(domainSpecWithIfacesResource.Devices.Interfaces) {
+	if len(domainSpec.Devices.Interfaces) == len(originalIfaces) {
 		return dom, nil
 	}
 
 	domainSpecWithoutIfacePlaceholders, err := util.GetDomainSpecWithFlags(dom, libvirt.DOMAIN_XML_INACTIVE)
 	if err != nil {
 		return nil, err
 	}
-	domainSpecWithoutIfacePlaceholders.Devices.Interfaces = domainSpec.Devices.Interfaces
+	domainSpecWithoutIfacePlaceholders.Devices.Interfaces = originalIfaces
 	// Only the devices are taken into account because some parameters are not assured to be returned when
 	// getting the domain spec (e.g. the `qemu:commandline` section).
 	domainSpecWithoutIfacePlaceholders.Devices.DeepCopyInto(&domainSpec.Devices)
 
 	return f(vmi, domainSpec)
 }
 
-func appendPlaceholderInterfacesToTheDomain(vmi *v1.VirtualMachineInstance, domainSpec *api.DomainSpec) *api.DomainSpec {
-	domainSpecWithIfacesResource := domainSpec.DeepCopy()
+func appendPlaceholderInterfacesToTheDomain(vmi *v1.VirtualMachineInstance, domainSpec *api.DomainSpec) []api.Interface {
+	originalIfaces := make([]api.Interface, len(domainSpec.Devices.Interfaces))
+	copy(originalIfaces, domainSpec.Devices.Interfaces)
+
 	interfacePlaceholderCount := ReservedInterfaces - len(vmi.Spec.Domain.Devices.Interfaces)
 	for i := 0; i < interfacePlaceholderCount; i++ {
-		domainSpecWithIfacesResource.Devices.Interfaces = append(
-			domainSpecWithIfacesResource.Devices.Interfaces,
+		domainSpec.Devices.Interfaces = append(
```

Comment:
This change causes the function to be mutable, having the side-effect of changing the inputted domain spec. I think we should avoid doing things like this as much as possible.

I prefer the input not to mutate and only the return value to have new values.
This is what I would expect from other objects/functions to behave like.